src/DevTools/MockRelayRenderer.tsx

```diff
@@ -0,0 +1,225 @@
+        '`jest.unmock("react-relay")`'
+    )
+  }
+
+  const network = createMockNetworkLayer({
+    Query: () => ({}),
+    ...mockResolvers,
+  })
+  const source = new RecordSource()
+  const store = new Store(source)
+  const environment = new Environment({
+    network,
+    store,
+  })
+
+  return (
+    <ContextProvider relayEnvironment={environment}>
+      <QueryRenderer
+        // tslint:disable-next-line relay-operation-generics
+        query={query}
+        environment={environment}
+        variables={{}}
+        render={({ error, props, retry }) => {
+          if (props) {
+            return <Component {...props} />
+          } else if (error) {
+            return <div className="relay-error">{error}</div>
+          } else {
+            return <div className={LoadingClassName}>Loading</div>
+          }
+        }}
+      />
+    </ContextProvider>
+  )
+}
+
+/**
+ * Renders a tree of Relay containers and resolves the returned promise once
+ * rendering has finished.
+ *
+ * It does this by checking the tree for the existence of an element with the
+ * class defined by `LoadingClassName` from the `renderWithLoadProgress` module.
+ * I.e. as long as at least 1 element exists in the tree with that class name,
+ * rendering is not considered finished.
+ *
+ * @param params
+ * See {@link MockRelayRenderer}
+ *
+ * @param until
+ * An optional callback that is used to test wether rendering should be
+ * considered finished. This is a regular enzyme wrapper.
+ * 
+ * @param wrapper
+ * An optional component that the Relay tree should be nested in. Use this to
+ * e.g. setup any context provider components etc.
+ *
+ * @example
+ *
+   ```tsx
+   jest.unmock("react-relay")
+ 
+   const Artwork = createFragmentContainer(
+     props => (
+       <div>
+         <span>{props.artwork.title}}</span>
+         <img src={props.artwork.image.url} />
+       </div>
+     ),
+     graphql`
+       fragment MockRelayRenderer_artwork on Artwork {
+         image {
+           url
+         }
+       }
+     `
+   )
+
+   it("renders a Relay tree", () => {
+     return renderRelayTree({
+       Component: Artwork,
+       query: graphql`
+         query MockRelayRendererQuery {
+           artwork(id: "mona-lisa") {
+             ...MockRelayRenderer_artwork
+           }
+         }
+       `,
+       mockResolvers: {
+         Artwork: () => ({
+           title: "Mona Lisa",
+           image: {
+             url: "http://test/image.jpg",
+           },
+         }),
+       },
+     }).then(wrapper => {
+       expect(wrapper.find("span").text()).toEqual("Mona Lisa")
+       expect(wrapper.find("img").props().src).toEqual("http://test/image.jpg")
+     })
+   })
+   ```
+ *
+ */
+export function renderRelayTree<
+  P = {},
+  S = {},
+  C extends React.Component = React.Component
+>(
+  params: MockRelayRendererProps & {
+    until?: UntilCallback<P, S, C>
```

Comment:
`UntilCallback` usage makes total sense but I'm wondering if there might be a slightly more descriptive name for it? Since it **a)** involves a common situation around nested `QueryRenderer`'s and **b)** could be used in other ways.

Reply:
Am I correct in understanding that you are suggesting to rename the parameters from `until` to `renderUntil`? And also rename the callback type from `UntilCallback` to `RenderUntilCallback`?

Reply:
With TS this isn't _so_ much of an issue because one can hover and read description, but not everyone uses TS -- so was thinking of some way of better describe what's happening here. `until` -> `waitUntil` -> `renderUntil`. Haha not too sure... perhaps the variety of its uses warrants a simple name.

Reply:
I didn’t go with `waitUntil` in the first place because this implementation is very tied to rendering React components and so didn’t want to be too ambiguous about that. I’ll rename things as per my above understanding for now.

Reply:
Sounds good -- i think `renderUntil` is a little more clear.